node/node.go

```diff
@@ -348,27 +348,6 @@ func NewNode(config *cfg.Config,
 	transport.AddChannelDescriptors(evReactorShim.GetChannels())
 	transport.AddChannelDescriptors(stateSyncReactorShim.GetChannels())
 
-	// setup Transport and Switch
-	sw := createSwitch(
-		config, transport, p2pMetrics, mpReactorShim, bcReactorForSwitch,
-		stateSyncReactorShim, csReactorShim, evReactorShim, proxyApp, nodeInfo, nodeKey, p2pLogger,
-	)
-
-	err = sw.AddPersistentPeers(strings.SplitAndTrimEmpty(config.P2P.PersistentPeers, ",", " "))
-	if err != nil {
-		return nil, fmt.Errorf("could not add peers from persistent-peers field: %w", err)
-	}
-
-	err = sw.AddUnconditionalPeerIDs(strings.SplitAndTrimEmpty(config.P2P.UnconditionalPeerIDs, ",", " "))
-	if err != nil {
-		return nil, fmt.Errorf("could not add peer ids from unconditional_peer_ids field: %w", err)
-	}
-
-	addrBook, err := createAddrBookAndSetOnSwitch(config, sw, p2pLogger, nodeKey)
-	if err != nil {
-		return nil, fmt.Errorf("could not create addrbook: %w", err)
-	}
-
```

Comment:
Good work here. I forgot that with my changes to the p2p channels we were no longer reliant on setting up the switch when using the new p2p stack. Hopefully there's no other parts of the code that depend on switch being non nil

Reply:
hopefully indeed! 

if there are it'd be better to panic so we can fix it :)